docs/time-varying-infectiousness.md

```diff
@@ -0,0 +1,205 @@
+`HashMap<PersonId, PlanId>`, removing the need to iterate through a vector and instead enabling us to
+directly use the `.insert` method on the `HashMap`.
+
+More generally, this idea applies to any case where we need to change the number of infection attempts
+part way through an infection course. Sequentially scheduling the attempts makes it possible to accomodate
+changes to the number of infection attempts that may happen in the middle of an infection course.
+
+Why not just check whether the agent is alive or not at the beginning of the infection attempt? If they are
+not alive, simply skip the infection attempt. There are two reasons this is not the desired solution. First,
+it is cleaner to handle removing an agent at the point at which they become no longer relevant rather than
+writing code that continuously checks for an agent's relevance. Secondly, there are other cases when the number
+of infection attempts may need to change partially through an infection. Imagine an agent becomes hospitalized
+after becoming infected. If contact rates are lower in the hospital, the number of contacts the agent may
+decrease. Clearly, explictly checking an agent's setting, alive status, etc. at the beginning of an infection
+attempt becomes not only cumbersome for many different settings but does not encourage modular separation
+of the transmission workflow from other parts of the model. On the other hand, sequentially scheduling infection
+attempts enables arbitrary changes to the effective contact rate to occur while an agent is infectious and have
+them be updated in the simplest way possible.
+
+### Rejection sampling from an arbitrary generation interval
+
+Let us consider a case where the infectiousness distribution changes over time. Concretely, imagine that
+infectiousness is a function of the viral load, and an agent gets an antiviral partially through their
+infection course. Depending on when the agent gets the antiviral, they have a different reduction in viral
+load. Therefore, even if we know that an agent will get an antiviral, we don't know their infectiousness
+distribution in advance.
+
+It is still straightforward to sample the appropriate infectiousness distribution without needing to change
+the generation interval. We can use [rejection sampling](https://en.wikipedia.org/wiki/Rejection_sampling).
+In this case, we still sample infection attempt times from the pre-antiviral infectiousness distribution, and
+we accept those sampled times as actual infection attempts with probability $a(t) / g(t)$ where $a(t)$ is the
+post-antiviral infectiousness distribution and $g(t)$ is the underlying pre-antiviral infectiousness
+distribution. Note that $a(t)$ and $g(t)$ must be on an absolute scale in this example and not scaled to have
+a unit integral. In the case where they are scaled, $g(t)$ can be rescaled to be $Mg(t)$ where $M = \max a(t)$.
+
+This general idea of rejection sampling is useful for other applications. Consider the case
```

Comment:
Not quite sure what the purpose of this paragraph is. I think the previous paragraph drives the message home just fine.